pkg/config/vlan.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2021 NVIDIA CORPORATION
+// Copyright 2018-2019 Red Hat, Inc.
+// Copyright 2014 CNI authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"errors"
+	"sort"
+
+	"github.com/Mellanox/accelerated-bridge-cni/pkg/types"
+)
+
+func splitVlanIds(trunks []types.Trunk) ([]int, error) {
+	vlans := make(map[int]bool)
+	for _, item := range trunks {
+		var minID, maxID, id int
+		if item.MinID != nil {
+			minID = *item.MinID
+			if minID < 0 || minID > 4096 {
+				return nil, errors.New("incorrect trunk minID parameter")
+			}
+		}
+		if item.MaxID != nil {
+			maxID = *item.MaxID
+			if maxID < 0 || maxID > 4096 {
+				return nil, errors.New("incorrect trunk maxID parameter")
+			}
+			if maxID < minID {
+				return nil, errors.New("minID is greater than maxID in trunk parameter")
+			}
+		}
+		if minID > 0 && maxID > 0 {
```

Comment:
what about overlapping vlans ? shouldnt we fail ?

Reply:
User input can contain overlapping vlans. Output is a sorted slice of unique values, e.g.  [4,10,35]

Reply:
makes sense, but i wonder if there is an overlap, it might hint towards user configuration error, perhaps we should emit a warning in this case.